debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java

```diff
@@ -279,6 +281,26 @@ public void shouldReceiveChangesForDeletes() throws Exception {
         assertEquals(topicName, record.topic());
         VerifyRecord.isValidTombstone(record, PK_FIELD, 2);
     }
+
+    @Test
+    public void shouldReceiveNumericTypeAsDouble() throws Exception {
+        PostgresConnectorConfig config = new PostgresConnectorConfig(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.DECIMAL_HANDLING_MODE, PostgresConnectorConfig.DecimalHandlingMode.DOUBLE)
+                .build());
+        PostgresTaskContext context = new PostgresTaskContext(config, new PostgresSchema(config));
+        recordsProducer = new RecordsStreamProducer(context, new SourceInfo(config.serverName()));
+
+        TestHelper.executeDDL("postgres_create_tables.ddl");
+
+        consumer = testConsumer(1);
+        recordsProducer.start(consumer);
+
+        List<SchemaAndValueField> schemasAndValuesForNumericType = schemasAndValuesForNumericType();
+        schemasAndValuesForNumericType.set(3, new SchemaAndValueField("d", Schema.OPTIONAL_FLOAT64_SCHEMA, 1.1d));
+        schemasAndValuesForNumericType.set(4, new SchemaAndValueField("n", Schema.OPTIONAL_FLOAT64_SCHEMA, 22.22d));
+
+        assertInsert(INSERT_NUMERIC_TYPES_STMT, schemasAndValuesForNumericType);
```

Comment:
@ssiahetiong, I'm getting an exception due to the mismatch of the `BigDecimal` value being passed which isn't expected by Connect for the FLOAT64 column.

I'm surprised that the proposed change doesn't contain a change to `PostgresValueConverter.convertDecimal()` which I think should emit a double to make happen what you intend.

Perhaps you can share some more insights on your testing?